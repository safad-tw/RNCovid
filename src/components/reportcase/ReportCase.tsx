import React, { useState } from 'react';
import {
  Text,
  View,
  TextInput,
  StyleSheet,
  TouchableOpacity, Alert
} from 'react-native';
import RNPickerSelect from 'react-native-picker-select';
import covidSummaryStore from '../../store/CovidSummaryStore'
import LanguagesStrings from './../../localization/LanguagesStrings';


const ReportCase = () => {

  const [numberOfCase, setNumberOfCases] = useState('')
  const [country, setCountry] = useState('')
  const [caseType, setCaseType] = useState('')

  const covidCases = covidSummaryStore((state) => state.covidCases.Countries);

  const updateCovid = covidSummaryStore((state) => state.updateCovid);

  const getCountries = () => {
    if (covidCases) {
      return covidCases.map((country: any) => (
        { 'label': country.Country, 'value': country.CountryCode }
      ));
    }
    return []
  }

  const pickerStyle: any = {
    inputIOS: {
      color: 'gray',
      paddingHorizontal: 10,
      backgroundColor: 'white',
      borderRadius: 20,
      margin: 5,
      borderColor: 'black',
      borderWidth: 1,
      height: 40,
      fontSize: 18,
      textAlign: 'center',
      fontFamily: "Avenir-medium"
    },
    placeholder: {
      color: 'gray',
      fontSize: 18,
      textAlign: 'center',
      fontFamily: "Avenir-medium",
    },
    inputAndroid: {
      color: 'gray',
      paddingHorizontal: 10,
      backgroundColor: 'white',
      borderRadius: 5,
      margin: 5,
      fontSize: 18,
      textAlign: 'center',
      fontFamily: "Avenir-medium",
    },
  };

  const caseTypes = [
    { label: 'Death', value: 'death' },
    { label: 'Recovery', value: 'recovery' },
    { label: 'Active', value: 'active' }
  ]

  const showAlert = (title:string,message:string) =>
    Alert.alert(
      title,
      message,
      [],
      {
        cancelable: true
      }
    );


  const submitClicked = () => {
    if (numberOfCase !== '' && country !== '' && caseType !== '') {
      updateCovid(caseType, country, Number(numberOfCase))
      showAlert(LanguagesStrings.success,LanguagesStrings.newCaseReported)
    } else {
      showAlert(LanguagesStrings.error,
      LanguagesStrings.enterAllTheFields)
    }
  };

  return (
    <View style={styles.container}>
      <Text style={styles.label}> Country: </Text>
      <RNPickerSelect
        style={pickerStyle}
        onValueChange={(value) => setCountry(value)}
        items={getCountries()}
      />

      <Text style={styles.label}> Case type: </Text>
      <RNPickerSelect
        style={pickerStyle}
        onValueChange={(value) => setCaseType(value)}
        items={caseTypes}
      />

      <Text style={styles.label}> Number of cases: </Text>
      <TextInput
        style={styles.textInput}
        keyboardType='numeric'
        placeholder={LanguagesStrings.enterNumberOfCase}
        onChangeText={(text) => setNumberOfCases(text)}
        value={numberOfCase}
      />
      <TouchableOpacity
        onPress={submitClicked}
        style={styles.submitButton}>
        <Text>Submit</Text>
      </TouchableOpacity>


    </View>
  )
}


const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingTop: 40,
    alignItems: "center"
  },
  textInput: {
    color: 'gray',
    width: '95%',
    paddingHorizontal: 10,
    backgroundColor: 'white',
    borderRadius: 20,
    margin: 5,
    borderColor: 'black',
    borderWidth: 1,
    height: 40,
    fontSize: 18,
    textAlign: 'center',
    fontFamily: "Avenir-medium"
  },
  label: {
    marginTop: 20,
    fontSize: 18,
    textAlign: 'center',
    fontFamily: "Avenir-black",
  },
  countryContainer: {
    height: 50,
    paddingTop: 40,
    width: '90%',
    alignItems: "center",
    borderRadius: 20,
    borderColor: 'black',
    borderWidth: 2,
    padding: 10,
    margin: 5,
  },
  submitButton: {
    marginTop: 20,
    width: '80%',
    height: 50,
    justifyContent: 'center',
    alignItems: 'center',
    padding: 10,
    borderRadius: 25,
    backgroundColor: '#ccc',
  },
});

export default ReportCase;
