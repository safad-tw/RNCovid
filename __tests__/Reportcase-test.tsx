import * as React from "react";
import renderer from "react-test-renderer";
import { render, fireEvent } from '@testing-library/react-native';
import '@testing-library/jest-dom'
import ReportCase from "../src/components/reportcase/ReportCase";


test("Render component", () => {
    const tree = renderer.create(<ReportCase/>).toJSON();
    expect(tree).not.toBeNull()
  });

  test('Renders snapshot as expected', () => {
    const tree = renderer.create(<ReportCase/>).toJSON();
    expect(tree).toMatchSnapshot();
  });