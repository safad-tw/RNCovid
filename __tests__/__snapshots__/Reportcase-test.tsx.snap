// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders snapshot as expected 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "paddingTop": 40,
    }
  }
>
  <Text
    style={
      Object {
        "fontFamily": "Avenir-black",
        "fontSize": 18,
        "marginTop": 20,
        "textAlign": "center",
      }
    }
  >
     Country: 
  </Text>
  <View
    style={
      Array [
        Object {
          "alignSelf": "stretch",
        },
        undefined,
      ]
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="ios_touchable_wrapper"
    >
      <View
        pointerEvents="box-only"
      >
        <TextInput
          editable={false}
          style={
            Array [
              Object {
                "backgroundColor": "white",
                "borderColor": "black",
                "borderRadius": 20,
                "borderWidth": 1,
                "color": "gray",
                "fontFamily": "Avenir-medium",
                "fontSize": 18,
                "height": 40,
                "margin": 5,
                "paddingHorizontal": 10,
                "textAlign": "center",
              },
              Object {
                "color": "gray",
                "fontFamily": "Avenir-medium",
                "fontSize": 18,
                "textAlign": "center",
              },
            ]
          }
          testID="text_input"
          value="Select an item..."
        />
      </View>
    </View>
    <Modal
      hardwareAccelerated={false}
      onOrientationChange={[Function]}
      supportedOrientations={
        Array [
          "portrait",
          "landscape",
        ]
      }
      testID="ios_modal"
      transparent={true}
      visible={false}
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flex": 1,
            "opacity": 1,
          }
        }
        testID="ios_modal_top"
      />
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#f8f8f8",
              "borderTopColor": "#dedede",
              "borderTopWidth": 1,
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "space-between",
              "paddingHorizontal": 10,
              "zIndex": 2,
            },
            undefined,
          ]
        }
        testID="input_accessory_view"
      >
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              undefined,
            ]
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderColor": "#a1a1a1",
                    "borderRightWidth": 1.5,
                    "borderTopWidth": 1.5,
                    "height": 15,
                    "width": 15,
                  },
                  undefined,
                  Object {
                    "marginLeft": 11,
                    "transform": Array [
                      Object {
                        "translateY": 4,
                      },
                      Object {
                        "rotate": "-45deg",
                      },
                    ],
                  },
                  undefined,
                  Object {},
                ]
              }
            />
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderColor": "#a1a1a1",
                    "borderRightWidth": 1.5,
                    "borderTopWidth": 1.5,
                    "height": 15,
                    "width": 15,
                  },
                  undefined,
                  Object {
                    "marginLeft": 22,
                    "transform": Array [
                      Object {
                        "translateY": -5,
                      },
                      Object {
                        "rotate": "135deg",
                      },
                    ],
                  },
                  undefined,
                  Object {},
                ]
              }
            />
          </View>
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          hitSlop={
            Object {
              "bottom": 4,
              "left": 4,
              "right": 4,
              "top": 4,
            }
          }
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="done_button"
        >
          <View
            testID="needed_for_touchable"
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#007aff",
                    "fontSize": 17,
                    "fontWeight": "600",
                    "paddingRight": 11,
                    "paddingTop": 1,
                  },
                  undefined,
                  Object {},
                ]
              }
              testID="done_text"
            >
              Done
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#d0d4da",
              "justifyContent": "center",
            },
            Object {
              "height": 215,
            },
            undefined,
          ]
        }
      >
        <View>
          <RNCPicker
            items={
              Array [
                Object {
                  "label": "Select an item...",
                  "testID": undefined,
                  "textColor": 4288585892,
                  "value": null,
                },
              ]
            }
            onChange={[Function]}
            selectedIndex={0}
            style={
              Array [
                Object {
                  "height": 216,
                },
                undefined,
              ]
            }
            testID="ios_picker"
          />
        </View>
      </View>
    </Modal>
  </View>
  <Text
    style={
      Object {
        "fontFamily": "Avenir-black",
        "fontSize": 18,
        "marginTop": 20,
        "textAlign": "center",
      }
    }
  >
     Case type: 
  </Text>
  <View
    style={
      Array [
        Object {
          "alignSelf": "stretch",
        },
        undefined,
      ]
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="ios_touchable_wrapper"
    >
      <View
        pointerEvents="box-only"
      >
        <TextInput
          editable={false}
          style={
            Array [
              Object {
                "backgroundColor": "white",
                "borderColor": "black",
                "borderRadius": 20,
                "borderWidth": 1,
                "color": "gray",
                "fontFamily": "Avenir-medium",
                "fontSize": 18,
                "height": 40,
                "margin": 5,
                "paddingHorizontal": 10,
                "textAlign": "center",
              },
              Object {
                "color": "gray",
                "fontFamily": "Avenir-medium",
                "fontSize": 18,
                "textAlign": "center",
              },
            ]
          }
          testID="text_input"
          value="Select an item..."
        />
      </View>
    </View>
    <Modal
      hardwareAccelerated={false}
      onOrientationChange={[Function]}
      supportedOrientations={
        Array [
          "portrait",
          "landscape",
        ]
      }
      testID="ios_modal"
      transparent={true}
      visible={false}
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flex": 1,
            "opacity": 1,
          }
        }
        testID="ios_modal_top"
      />
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#f8f8f8",
              "borderTopColor": "#dedede",
              "borderTopWidth": 1,
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "space-between",
              "paddingHorizontal": 10,
              "zIndex": 2,
            },
            undefined,
          ]
        }
        testID="input_accessory_view"
      >
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              undefined,
            ]
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderColor": "#a1a1a1",
                    "borderRightWidth": 1.5,
                    "borderTopWidth": 1.5,
                    "height": 15,
                    "width": 15,
                  },
                  undefined,
                  Object {
                    "marginLeft": 11,
                    "transform": Array [
                      Object {
                        "translateY": 4,
                      },
                      Object {
                        "rotate": "-45deg",
                      },
                    ],
                  },
                  undefined,
                  Object {},
                ]
              }
            />
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderColor": "#a1a1a1",
                    "borderRightWidth": 1.5,
                    "borderTopWidth": 1.5,
                    "height": 15,
                    "width": 15,
                  },
                  undefined,
                  Object {
                    "marginLeft": 22,
                    "transform": Array [
                      Object {
                        "translateY": -5,
                      },
                      Object {
                        "rotate": "135deg",
                      },
                    ],
                  },
                  undefined,
                  Object {},
                ]
              }
            />
          </View>
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          hitSlop={
            Object {
              "bottom": 4,
              "left": 4,
              "right": 4,
              "top": 4,
            }
          }
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="done_button"
        >
          <View
            testID="needed_for_touchable"
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#007aff",
                    "fontSize": 17,
                    "fontWeight": "600",
                    "paddingRight": 11,
                    "paddingTop": 1,
                  },
                  undefined,
                  Object {},
                ]
              }
              testID="done_text"
            >
              Done
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#d0d4da",
              "justifyContent": "center",
            },
            Object {
              "height": 215,
            },
            undefined,
          ]
        }
      >
        <View>
          <RNCPicker
            items={
              Array [
                Object {
                  "label": "Select an item...",
                  "testID": undefined,
                  "textColor": 4288585892,
                  "value": null,
                },
                Object {
                  "label": "Death",
                  "testID": undefined,
                  "textColor": undefined,
                  "value": "death",
                },
                Object {
                  "label": "Recovery",
                  "testID": undefined,
                  "textColor": undefined,
                  "value": "recovery",
                },
                Object {
                  "label": "Active",
                  "testID": undefined,
                  "textColor": undefined,
                  "value": "active",
                },
              ]
            }
            onChange={[Function]}
            selectedIndex={0}
            style={
              Array [
                Object {
                  "height": 216,
                },
                undefined,
              ]
            }
            testID="ios_picker"
          />
        </View>
      </View>
    </Modal>
  </View>
  <Text
    style={
      Object {
        "fontFamily": "Avenir-black",
        "fontSize": 18,
        "marginTop": 20,
        "textAlign": "center",
      }
    }
  >
     Number of cases: 
  </Text>
  <TextInput
    keyboardType="numeric"
    onChangeText={[Function]}
    placeholder="Enter number of cases"
    style={
      Object {
        "backgroundColor": "white",
        "borderColor": "black",
        "borderRadius": 20,
        "borderWidth": 1,
        "color": "gray",
        "fontFamily": "Avenir-medium",
        "fontSize": 18,
        "height": 40,
        "margin": 5,
        "paddingHorizontal": 10,
        "textAlign": "center",
        "width": "95%",
      }
    }
    value=""
  />
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ccc",
        "borderRadius": 25,
        "height": 50,
        "justifyContent": "center",
        "marginTop": 20,
        "opacity": 1,
        "padding": 10,
        "width": "80%",
      }
    }
  >
    <Text>
      Submit
    </Text>
  </View>
</View>
`;
